### PR TITLE
DTSPO-4613 - bump azuread provider

### PR DESCRIPTION
**Pull request by endakelly**
- Adding azuread provider
- Bumping azuread provider version

### CODE REVIEW

File: infrastructure/state.tf [hunk 1]
```diff
@@ -11,7 +11,7 @@ terraform {
     }
     azuread = {
       source  = "hashicorp/azuread"
-      version = "1.6.0"
+      version = "~> 2.0.0"
```
Comment (timja) [inline infrastructure/state.tf:14]: best to fix the version to latest working

Comment (endakelly) [reply infrastructure/state.tf:14]: Not sure what you mean. The provider wasn't explicitly in the state.tf until now so I added it to get round the terraform show error we've been seeing recently. Assuming it needs 2.0.0 going by the error in the latest pipeline run. 

`Could not retrieve the list of available versions for provider
hashicorp/azuread: no available releases match the given constraints 1.6.0,>= 2.0.0`

Comment (timja) [reply infrastructure/state.tf:14]: be explicit:
```suggestion
      version = "2.5.0"
```

Comment (endakelly) [reply infrastructure/state.tf]: Updated

**Review by timja — APPROVED** [review]

**Pull request merged**